CMakeLists.txt

```diff
@@ -297,10 +297,7 @@ set(
   BOOST_CHRONO_DONT_PROVIDES_DEPRECATED_IO_SINCE_V2_0_0
   BOOST_STACKTRACE_USE_ADDR2LINE
   BOOST_ASIO_NO_DEPRECATED
-  # Historically, iRODS has used OpenSSL API version 1.1.x. OpenSSL 3.x deprecated and even removed some API calls
-  # in use by iRODS. This requires a compile definition which specifies the API version with which the target
-  # should be compatible. For more information, see https://www.openssl.org/docs/man3.0/man7/OPENSSL_API_COMPAT.html
-  OPENSSL_API_COMPAT=10100
+  OPENSSL_NO_DEPRECATED
```

Comment:
Let's change this to `OPENSSL_NO_DEPRECATED_3_0`. I don't want to make it difficult to build against future versions of OpenSSL, and `OPENSSL_NO_DEPRECATED_3_0` ensures we don't use anything prior to 3.0. Should we happen to use or reference something deprecated in 3.1+, I think a compiler warning/error (depending on build environment) will suffice.

Reply:
Done